backend/utils/S3Uploader.ts

```diff
@@ -44,7 +46,67 @@ const UploadToS3 = async (file: Express.Multer.File, path: string) => {
     console.log("Local file deleted successfully.");
   } catch (error) {
     console.log("Error has occured in the S3 Uploader", error);
+    throw new AppError(
+      AppErrorName.FILE_UPLOAD_ERROR,
+      "Error occurred while uploading file to S3",
+      500,
+      true,
+    );
   }
 };
 
+const deleteFromS3 = async (path: string) => {
+  try {
+    // AWS Credentials and bucket information
+    const accessKeyId = env.AWS_ACCESS_KEY_ID;
+    const secretAccessKey = env.AWS_SECRET_ACCESS_KEY;
+    const region = env.AWS_REGION;
+    const Bucket = env.AWS_BUCKET_NAME;
+
+    // Initialize S3 Client
+    const s3Client = new S3Client({
+      credentials: {
+        accessKeyId,
+        secretAccessKey,
+      },
+      region,
+    });
+
+    // Create and send a delete object command
+    const deleteCommand = new DeleteObjectCommand({
+      Bucket,
+      Key: path,
+    });
+
+    const response = await s3Client.send(deleteCommand);
+    console.log("File deleted successfully from S3.");
+  } catch (error) {
+    console.error("Error occurred while deleting file from S3:", error);
+    throw new AppError(
+      AppErrorName.FILE_DELETE_ERROR,
+      "Error occurred while deleting file from S3",
+      500,
+      true,
+    );
+  }
+};
+
+const generateUniqueFileName = (originalName: string, id: string) => {
+  if (originalName.length > 100) {
+    originalName = originalName.substring(0, 100);
+  }
+
+  // Create a timestamp
+  const timestamp = new Date().toISOString().replace(/:/g, "-");
+
+  // Extract the file extension
+  const extension = originalName.split(".").pop();
+
+  // Construct the unique file name
+  return `${id}-${timestamp}_${originalName}`;
```

Comment:
Maybe use something like an underscore between id and timestamp instead of a dash to make it clear where UUID ends and timestamp begins?
right now it looks like: `images/events/b4a5dd11-4b98-4ec8-8466-9bc0328267df-2024-01-29T19-27-19.044Z_hockey.jpg`